cmake/HPX_AddConfigTest.cmake

```diff
@@ -384,6 +384,14 @@ macro(hpx_check_for_cxx11_std_array)
 endmacro()
 
 ###############################################################################
+macro(hpx_check_for_cxx11_std_atomic)
+  add_hpx_config_test(HPX_WITH_CXX11_ATOMIC
+    SOURCE cmake/tests/cxx11_std_atomic.cpp
+    LIBRARIES "-latomic"
```

Comment:
I guess it is fine as long as the compiler only warns for unknown arguments and doesn't fail if it can't find the library to link against.